controllers/ovndbcluster_controller.go

```diff
@@ -719,7 +719,11 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 	// cluster member so it can be resolved from outside cluster (edpm nodes)
 	if instance.Spec.NetworkAttachment != "" {
 		var dnsIPsList []string
-		for _, ovnPod := range podList.Items[:*(instance.Spec.Replicas)] {
+		for it, ovnPod := range podList.Items[:*(instance.Spec.Replicas)] {
```

Comment:
I don't think you should access podList.Items past its length, even if it doesn't crash at this point.

Can't you compare lengths before iteration?

Reply:
and if you'd like to iterate through the list of "pods that are supposed to run", you could iterate over `podList.Items[:min(Replicas, len(podList.Items))]` (that's pseudo-code).

Reply:
Done